src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1466,6 +1572,67 @@ public async Task sendAsync_ConcurentSendReceive_Ok(bool shouldWait, HttpStatusC
             });
         }
 
+        [Fact]
+        public async Task SendAsync_ConcurentSendReceive_Fail()
+        {
+            TaskCompletionSource<bool> tsc = new TaskCompletionSource<bool>();
+            string requestContent = new string('*', 300);
+            const string responseContent = "SendAsync_ConcurentSendReceive_Fail";
+            var stream = new CustomContent.SlowTestStream(Encoding.UTF8.GetBytes(requestContent), tsc, trigger:1, count: 10);
+
+            await Http2LoopbackServer.CreateClientAndServerAsync(async url =>
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Post, url);
+                    request.Version = new Version(2,0);
+                    request.Content = new StreamContent(stream);
+
+                    // This should fail either while getting response headers or while reading response body.
+                    HttpResponseMessage response = await client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+                    // Wait for request body start streaming.
+                    await tsc.Task.ConfigureAwait(false);
+                    // and inject distinct exception on ReqeuestStream.
+                    stream.SetException(new ArithmeticException("Injected test exception"));
+                    try
+                    {
+                         string responseBody = await response.Content.ReadAsStringAsync();
+                         Assert.True(false, "Should not be here");
+                    }
+                    catch (HttpRequestException e)
```

Comment:
AFAIR Assert.Throws returns thrown exception

Reply:
Both Assert.Throws<T>() and Assert.ThrowsAsync<T>() give back the exception as the return.